[docs] Account Types: Remove cannot rename reference & other improvements

amandeepmittal opened the pull request:
# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub issues, forums posts, or feature requests.
-->

Refs ENG-5757

# How

<!--
How did you build this feature or fix this bug and why?
-->

This PR:
- removes info that organization account cannot be renamed
- updates and improves verbiage
- adds screenshots for organization account settings

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.
-->

All changes applied have been tested locally.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

amandeepmittal committed:
[docs] Account Types: Remove cannot rename reference & other improvements

amandeepmittal committed:
Improve this document and add some screenshots

kbrandwijk commented on docs/pages/accounts/account-types.md:
```diff
@@ -53,19 +69,19 @@ We have taken a lot of care to make sure that all of the functionality that you
+- New publish for a renamed project must be on SDK 43 or higher.
 
-> 💡 In the case where a project on your Personal or Organization Account (`source`) is sold/given to another company/person (receiving party) and the receiving party does not want to grant you Owner access to their `destination` account, you can create a new Organization Account (`escrow`), grant the receiving party Owner access, and transfer the project to the `escrow` account. The receiving par ...
+> In a scenario where a project on your Personal or Organization Account (`source`) is sold/given to another company/person (receiving party), and the receiving party does not want to grant you Owner access to their `destination` account, you can create a new Organization Account (`escrow`), grant the receiving party Owner access, and transfer the project to the `escrow` account. The receiving pa ...
```
I would simplify this a lot: 'If you want to transfer ownership of a project, but you can't be given Admin permissions on the destination account, you can create an escrow account etc. etc.'

kbrandwijk commented on docs/pages/accounts/account-types.md:
```diff
@@ -53,19 +69,19 @@ We have taken a lot of care to make sure that all of the functionality that you
 
-- The person performing the transfer must have "Owner" role on both the source and destination accounts.
-- New publishes for renamed project must be on SDK 43 or higher.
+- The person performing the transfer must have the Owner role on both the source and destination accounts.
```
I believe this has changed to 'Admin'

kelley-sharp replied:
What @kbrandwijk means is that both Owners and Admins can transfer projects (and rename orgs) now.

amandeepmittal committed:
Update docs/pages/accounts/account-types.md

Co-authored-by: Kim Brandwijk <kim.brandwijk@gmail.com>

kelley-sharp commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,48 +2,64 @@
+Personal Accounts are referred to as the default account type in the scenarios where you take action, but the activity does not specify which account it should belong.
 
-> This account is just for you, **never share access to this account for any reason.**
+> This account is just for you. Never share access to this account for any reason.
```
```suggestion
> You should not share authentication credentials for your Personal Accounts with anyone for any reason. Members will login to their own Personal Account to get access to any Team or Org Accounts they are a member of. Accounts belonged to can be accessed from the dropdown in the top left corner of the portal on [Expo.dev](https://expo.dev). 
```

kelley-sharp commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,48 +2,64 @@
 
-Every user of Expo has their own Personal Account. With this account, we can safely identify you as the developer or owner of an app. You can use Personal Accounts for all features Expo has to offer. In cases where you take an action that doesn't specify which account it should belong to, it will default to your Personal Account
+When you [create a new account](https://expo.dev/signup) with Expo, you are creating a Personal Account. This account is your individual account. Expo uses this account to safely identify you as your application's developer or owner. A Personal Account has access to all features provided by Expo.
+
```
```suggestion
## Personal Accounts
```

kelley-sharp commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,48 +2,64 @@
+| **Release bug fixes with updates**                                  | ✅                | ✅           |
+| **Transfer control of individual projects to another user**         | ✅                | ✅           |
+| **Transfer control of all projects to another user**                |                   | ✅           |
+| **Programmatic access with limited privileges**                     |                   | ✅           |
```
I really like the idea of a table, however I don't think it works since the language around Project Transfers is misleading. Also Team Accounts is missing. There seems to be some misunderstanding of the differences between accounts. See my comment on line 13. We might just want to delete the table.

kelley-sharp reviewed (changes requested):
I love the addition of images! @amandeepmittal Let me know if you want to pair on this page

amandeepmittal commented:
Sorry about the misunderstanding @kelley-sharp @kbrandwijk from my end.

I originally changed the "removes info that organization account cannot be renamed", added some screenshots for those settings, and updated the copy to make it more understandable.

I was not aware that you both are not aware of the current state of the document. The original copy exists in the docs: https://docs.expo.dev/accounts/account-types/ 

The changes I made to the copy were to follow our style guide and update the verbiage rather than adding any new technical info. 

Though, the points you both mentioned must be addressed as I sense there is some misinformation and missing information. For now (after discussing with Kelley), I am going to open a separate PR that just tackles the original info. I will keep this PR as a reference to continue working on improving this docs page. I guess a good thing that came out of the PR review for me was to be aware that this document is outdated :)

amandeepmittal commented:
Closing this PR since a draft is already a work in progress.
